walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java

```diff
@@ -183,4 +183,8 @@ public static TimeRange fromStartEnd(int start, int end, boolean inclusive) {
   public static TimeRange fromStartDuration(int start, int duration) {
     return new TimeRange(start, duration);
   }
+
+  public static TimeRange mergeTimeRanges(TimeRange first, TimeRange second) {
```

Comment:
Because you are creating a generic library "mergeTimeRanges" which could be used by other people, it should work as how people usually expect it to work. Your method is rather specific to your algorithm. I'd do:
1. if first and second don't overlap, throw an IllegalArgumentException. 
2. return a true merged range (min(starts), max(ends)).

Reply:
For my code, I want it to be able to add the correct TimeRange to List my depending if the two ranges either (1) overlap but the first doesn't contain the next range or (2) overlap and the first does contain the entire next range. The purpose of this code is so that I can "merge"/return the proper TimeRange when they do overlap. At the point when I call these functions, then it's guaranteed that first and second overlap. Could I just make it so that my function title is more specific then so that its just for this situation?

Reply:
So it's your choice. You can either make this general method here and use it, or create a more specific method under your class (not in this class) with a specific title and detailed comments on what it does and what the restrictions are. The former would make the code more readable and the latter would make the code slightly more efficient (one less comparison). Usually Google prefers readability over efficiency but in this case it's your call.

Reply:
I see. For the sake of time, I just went with the latter option. But for future reference, I'll focus on the internal preference for readability.